.github/workflows/validate-rules-syntax.yaml

```diff
@@ -0,0 +1,100 @@
+on: pull_request
+permissions:
+  contents: read
+  pull-requests: read
+jobs:
+  validate_loki_recording_rules:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Create tmp dir 
+        run: mkdir -p validation/recording-rules
+
+      - name: Generate Loki recording rules
+        run: yq e '.spec' loki/recording-rules.yaml > validation/recording-rules/rules.yaml
+
+      - name: Lint rules
+        uses: grafana//cortex-rules-action@v0.8.0
+        env:
+          ACTION: lint
```

Comment:
> lint
>
> Lints a rules file(s). The linter's aim is not to verify correctness but to fix YAML and PromQL expression formatting within the rule file(s). The linting happens in-place within the specified file(s). Does not interact with your Cortex cluster.

I'm not sure linting here makes sense?

Reply:
However it won't hurt IMO.

Reply:
It may: e.g. if it reformats the file; then the following `check` will have warnings with the wrong line numbers.